airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -270,7 +270,7 @@ def _retry(
             if original_exception:
                 raise original_exception
             raise DefaultBackoffException(request=request, response=response)
-        if response:
+        if response is not None:
```

Comment:
@BenoitHugonnard can you explain how this will solve the problem with 429? The change you made doesn't affect any logic.
`if response` is equal to `if response is not None`
If I'm not wrong :p

Reply:
Actually it is the key !

The issue we currently have, is when Zendesk responds with a 429 for organizations, the retry that can only happen `if response` is never called so Airbyte doesn't wait x seconds before retrying so Uncaught Error (see the associated issue).

The issue is that when we bool check the 429 response we get false but the content is not empty (we still get the headers having the info for the x-retry-after).

If we do this check you'll see :

```
for i in range(15):
    resp = requests.get(
        url="https://*****.zendesk.com/api/v2/organizations?start_time=946684800&page=1011",
        headers=headers,
        auth=auth
    )
    status_code = resp.status_code
    is_response = True if resp is not None else False
    print(status_code, is_response)

>>>200 True
200 True
200 True
200 True
200 True
200 True
200 True
200 True
200 True
200 True
429 True
429 True
429 True
429 True
429 True
```

Reply:
https://stackoverflow.com/questions/48347290/why-does-requests-response-object-bool-check-for-200-status-400

Reply:
Thanks for the info @BenoitHugonnard